File: C#/EncryptionAlgorithmsPerformance/DES.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Security.Cryptography;
using System.Text;
using System.Threading.Tasks;

namespace EncryptionAlgorithmsPerformance
{
    class DES
    {
        public static DESCryptoServiceProvider objDes = new DESCryptoServiceProvider();
        public static double encryptionTime;
        public static double decryptionTime;
        public static double encryptionCpuLoad;
        public static double decryptionCpuLoad;

        DES()
        {

        }



        public static void encrypt()
        {


           
            ICryptoTransform encryptor = objDes.CreateEncryptor();
            objDes.Padding = PaddingMode.Zeros;
            objDes.Mode = CipherMode.ECB;


            byte[] inputBytes = File.ReadAllBytes("textfile.txt");
            long startTime = nanoTime();
            byte[] encryptedBytes = encryptor.TransformFinalBlock(inputBytes, 0, inputBytes.Length);


            long endTime = nanoTime();
            FileStream fs = new FileStream("encryptedfile.txt", FileMode.Create, FileAccess.Write);
            fs.Write(encryptedBytes, 0, encryptedBytes.Length);
           
            encryptionTime = (endTime - startTime) / 10;
            encryptionTime /= 100;
            fs.Close();




        }

        public static void decrypt()
        {


            
            ICryptoTransform decryptor = objDes.CreateDecryptor();
            objDes.Padding = PaddingMode.Zeros;
            objDes.Mode = CipherMode.ECB;



            byte[] inputBytes = File.ReadAllBytes("encryptedfile.txt");
            long startTime = nanoTime();
            byte[] decryptedBytes = decryptor.TransformFinalBlock(inputBytes, 0, inputBytes.Length);


            long endTime = nanoTime();
            FileStream fs = new FileStream("decryptedfile.txt", FileMode.Create, FileAccess.Write);
            fs.Write(decryptedBytes, 0, decryptedBytes.Length);
            

            decryptionTime = (endTime - startTime) / 10;
            decryptionTime /= 100;
            fs.Close();



        }
        public static long nanoTime()
        {
            long nano = 10000L * Stopwatch.GetTimestamp();
            nano /= TimeSpan.TicksPerMillisecond;
            nano *= 100L;
            return nano;
        }

    }
}

File: Java/src/encryptionalgorithmsperformance/DES.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package encryptionalgorithmsperformance;

/**
 *
 * @author USER
 */
import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.lang.management.ManagementFactory;
import java.lang.management.ThreadMXBean;
import java.security.AccessControlException;
import javax.crypto.Cipher;
import javax.crypto.SecretKey;
import javax.crypto.spec.IvParameterSpec;
import javax.crypto.spec.SecretKeySpec;

/**
 *
 * @author USER
 */
public class DES {
    private static SecretKeySpec secretKey;
    private static byte[] key;
    static double encryptionTime;
    static double decryptionTime;
    static double encryptionCpuLoad;
    static double decryptionCpuLoad;
    DES()
    {
    	
    }
 
    
 
    public static void encrypt(File inputFile,File outputFile, SecretKey key)
    {
        
        try
        {
            ThreadMXBean newBean = ManagementFactory.getThreadMXBean();
            if (newBean.isThreadCpuTimeSupported())
            {
            newBean.setThreadCpuTimeEnabled(true);
            }
            else
            {
                throw new AccessControlException("");
            }
            long startThreadTime = newBean.getCurrentThreadCpuTime();
            Cipher cipher = Cipher.getInstance("DES/ECB/PKCS5Padding");
            SecretKeySpec keySpec = new SecretKeySpec(key.getEncoded(),"DES");
            cipher.init(Cipher.ENCRYPT_MODE, keySpec);
            
            FileInputStream inputStream = new FileInputStream(inputFile);
            byte[] inputBytes = new byte[(int) inputFile.length()];
            inputStream.read(inputBytes);
            long startTime = System.nanoTime();

            byte[] outputBytes = cipher.doFinal(inputBytes);


            FileOutputStream outputStream = new FileOutputStream(outputFile);
            outputStream.write(outputBytes);
            long endTime = System.nanoTime();
            inputStream.close();
            outputStream.close();
            long endthreadTime = newBean.getCurrentThreadCpuTime();
            encryptionCpuLoad = (endthreadTime - startThreadTime) / (double)(1000000);
            encryptionTime = (endTime - startTime)/(double)1000;
            
        }
       
        catch (AccessControlException e)
        {
            System.out.println("CPU Usage monitoring is not available!");
            System.exit(0);
        }
        catch (Exception e)
        {
            System.out.println("Error while encrypting: " + e.toString());
        }
    }
 
    public static void decrypt(File inputFile,File outputFile, SecretKey key)	
    {
        try
        {
            ThreadMXBean newBean = ManagementFactory.getThreadMXBean();
            if (newBean.isThreadCpuTimeSupported())
            {
            newBean.setThreadCpuTimeEnabled(true);
            }
            else
            {
                throw new AccessControlException("");
            }
            long startThreadTime = newBean.getCurrentThreadCpuTime();
            Cipher cipher = Cipher.getInstance("DES/ECB/PKCS5Padding");
            SecretKeySpec keySpec = new SecretKeySpec(key.getEncoded(),"DES");

            cipher.init(Cipher.DECRYPT_MODE, keySpec);
            
            FileInputStream inputStream = new FileInputStream(inputFile);
            byte[] inputBytes = new byte[(int) inputFile.length()];
            inputStream.read(inputBytes);
            long startTime = System.nanoTime(); 
            byte[] outputBytes = cipher.doFinal(inputBytes);


            FileOutputStream outputStream = new FileOutputStream(outputFile);
            outputStream.write(outputBytes);
            long endTime = System.nanoTime();
            long endthreadTime = newBean.getCurrentThreadCpuTime()+1;
            decryptionCpuLoad = (endthreadTime - startThreadTime) / (double)(1000000);
            decryptionTime = (endTime - startTime)/(double)1000;
     

        inputStream.close();
        outputStream.close();
            
        }
        catch (Exception e)
        {
            System.out.println("Error while decrypting: " + e.toString());
        }
    }
}